rspace/src/main/scala/coop/rchain/rspace/Trie.scala

```diff
@@ -0,0 +1,25 @@
+package coop.rchain.rspace
+
+import scodec.Codec
+import scodec.codecs._
+
+sealed trait Trie[+K, +V]                         extends Product with Serializable
+final case class Leaf[K, V](key: K, value: V)     extends Trie[K, V]
+final case class Node(pointerBlock: PointerBlock) extends Trie[Nothing, Nothing]
```

Comment:
I am looking at this encoding of the `Trie` and can't think of how `Leaf`s and `Node`s could be connected. Is the `value` in the `Leaf` becoming `Node` ? But then the naming is confusing.